### PR TITLE
Feature/processing data in models

### PR DESCRIPTION
**Pull request by eduardosancho**
# What was done in this PR by @eduardosancho 🎃 : 

### To play with the new features, clone this repo and run the following commands first:
1. `rails db:drop`
2. `rails db:create`
3. `rails db:migrate`
4. `rails db:seed`

This will populate the database with the default entries, so you can play with the queries, which will be mentioned specifically on their corresponding bullet point.

## Objectives:
- [x] Set up associations between models, following [this diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). 🔗 
- [x] Write SQL queries with Active Record. 🧲 

## Changes:
* Rename foreign_key columns in tables `users`, `comments`, and `likes` from **user_id** to **author_id**. ✏️ 
* Add `on_delete: :cascade` to all foreign_keys to enable destroying entries of the database despite of having associations. 👿 

## Additions:
* Implement models for `User`, `Post`, `Comment`, and `Like` with their corresponding associations from the [diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)
* Prepare `db/seeds.rb` to populate the database with default entries. 🌱 
* Enable all CRUD methods for all models [`create`,  `find`, `update`, `delete`]. Although `destroy` is recommended instead of `delete`. 🚀 
* Add method `recent_posts` that returns the 3 most recent posts for a given user.
* Add private method `update_posts_counter` that updates the posts_counter for a user.
* Add method `recent_comments` that returns the 5 most recent comments for a given post.
* Add private method `update_comments_counter` that updates the comments_counter for a post.
* Add private method `update_likes_counter` that updates the likes_counter for a post.
* Create a method called `current_user` that returns the first user from the database. 🥇 
<hr>

### Feel free to use the following commands to test the new features. Remember to first do `rails c`.
~~~ irb
User.first.recent_posts
User.create(name: 'Mary')
mary = User.find_by(name: 'Mary')
mary.update(bio: 'Micronaut')
Post.create(author: mary, title: 'My name is Mary', text: 'And I am a micronaut')
mary.posts # see all posts from mary
mary.posts_counter
Comment.create(post: Post.first, author: mary, text: 'I loved this article')
mary.comments # see all comments from mary
Post.first.recent_comments
Like.create(post: Post.first, author: mary)
mary.likes # see all likes from mary
Post.first.comments_counter
Post.first.likes_counter
User.find_by(bio: 'Micronaut').destroy
~~~

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  before_save :update_comments_counter
```
Comment (descholar-ceo) [inline app/models/comment.rb]: - [x] Just to make sure that we increment the counter after a successful save, we should use the `after_save` helper method, so I would suggest you do this insterad:
```suggestion
  after_save :update_comments_counter
```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  before_save :update_likes_counter
```
Comment (descholar-ceo) [inline app/models/like.rb]: - [x] Just to make sure that we increment the counter after a successful save, we should use the `after_save` helper method, so I would suggest you do this instead:
`````suggestion
  after_save :update_comments_counter
```

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @eduardosancho , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
### Hi @eduardosancho, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**